packages/repack/src/webpack/utils/getResolveOptions.ts

```diff
@@ -17,35 +35,83 @@
  *
  *   return {
  *     resolve: {
- *       ...Repack.getResolveOptions(platform),
+ *       ...Repack.getResolveOptions(platform, {
+ *         enablePackageExports: false,
+ *         preferNativePlatform: true
+ *       }),
  *     },
  *   };
  * };
  * ```
  */
-export function getResolveOptions(platform: string) {
+
+export function getResolveOptions(platform: string, options?: ResolveOptions) {
+  const preferNativePlatform = options?.preferNativePlatform ?? true;
+  const enablePackageExports = options?.enablePackageExports ?? false;
+
+  let extensions = ['.ts', '.js', '.tsx', '.jsx', '.json'];
```

Comment:
Shouldn't we match default Metro extensions order too?
https://github.com/facebook/metro/blob/7250388527dab827ba86f57b26ca5d8dd0098889/packages/metro-config/src/defaults/defaults.js#L56

Reply:
nice find 🚀 , it's best that we match that as well, it's a rare edge case but might cause issues nonetheless

Reply:
also we could re-use https://github.com/facebook/metro/blob/7250388527dab827ba86f57b26ca5d8dd0098889/packages/metro-config/src/defaults/defaults.js#L18-L52 here:
https://github.com/callstack/repack/blob/b0b8613eab82641e91607e0931eafc2360ff42a2/packages/repack/src/webpack/utils/assetExtensions.ts#L2-L43

Reply:
@szymonrybczak could you elaborate more? do you mean plugging into `@react-native/metro-config` ?

Reply:
ah, I misunderstood first comment. yeah, that's what I meant but that's not possible 😅

Reply:
the order of extensions is now matched exactly as in metro 👍